src/adapter/src/coord.rs

```diff
@@ -6546,7 +6532,7 @@ pub mod fast_path_peek {
                         compute_instance,
                         // Disable compaction by using None as the compaction window so that nothing
                         // can compact before the peek occurs below.
-                        None,
+                        Some(ReadPolicy::ValidFrom(Antichain::from_elem(timestamp))),
```

Comment:
You can probably still use `None` here. The timeline read hold will prevent the dataflow from compaction before the peek occurs.

Reply:
Is that guaranteed to be the case somehow in the code, or is it a side effect of the current implementation? Making this `None` would be good because then we can completely remove this argument.

Reply:
I think mostly a side effect of the current implementation.